operators/video_streaming/streaming_client_enhanced/streaming_client.cpp

```diff
@@ -0,0 +1,1601 @@
+          auto bgra_tensor_data = std::shared_ptr<uint8_t[]>(new uint8_t[bgra_frame_size]);
+          if (bgra_tensor_data) {
+            // Copy the BGRA data
+            std::memcpy(bgra_tensor_data.get(), bgra_buffer->data(), bgra_frame_size);
+
+            // Wrap the memory in the tensor
+            bgra_tensor_handle.value()->wrapMemory(
+                bgra_shape,
+                element_type,
+                element_size,
+                nvidia::gxf::ComputeTrivialStrides(bgra_shape, element_size),
+                nvidia::gxf::MemoryStorageType::kSystem,
+                bgra_tensor_data.get(),
+                [bgra_tensor_data](void*) mutable {
+                  // The shared_ptr will automatically clean up when it goes out of scope
+                  bgra_tensor_data.reset();
+                  return nvidia::gxf::Success;
+                });
+
+            // Emit the BGRA tensor entity to HoloViz
+            op_output.emit(maybe_bgra_entity.value(), "output_frames");
+            HOLOSCAN_LOG_INFO("✅ Emitted BGRA tensor for HoloViz: [{}x{}x4], {} bytes",
+                             expected_height, expected_width, bgra_frame_size);
+          } else {
+            HOLOSCAN_LOG_ERROR("Failed to allocate memory for BGRA tensor");
+          }
+        } else {
+          HOLOSCAN_LOG_ERROR("Failed to add BGRA tensor to entity");
+        }
+      } else {
+        HOLOSCAN_LOG_ERROR("Failed to create BGRA entity for HoloViz");
+      }
+    } catch (const std::exception& e) {
+      HOLOSCAN_LOG_ERROR("Exception creating BGRA tensor output: {}", e.what());
+    }
+
+    // Add retry logic for frame sending
+    bool send_success = false;
+    int send_attempts = 0;
+    const int max_send_attempts = 3;
+
+    while (!send_success && send_attempts < max_send_attempts) {
+      try {
+        // Check if client is ready to send frames (both streaming and upstream ready)
+        if (client_ && client_->isStreaming() && client_->isUpstreamReady()) {
+          HOLOSCAN_LOG_INFO("Attempting to send frame: {}x{}, {} bytes, "
+                            "streaming=active, upstream=ready",
+                            frame.getWidth(), frame.getHeight(), frame.getDataSize());
+
+          // Add detailed pre-send validation
+          HOLOSCAN_LOG_DEBUG("Pre-send validation:");
+          HOLOSCAN_LOG_DEBUG("  - Frame valid: {}", frame.isValid() ? "YES" : "NO");
+          HOLOSCAN_LOG_DEBUG("  - Frame data: {}", frame.getData() ? "VALID" : "NULL");
+          HOLOSCAN_LOG_DEBUG("  - Frame size: {} bytes", frame.getDataSize());
+          HOLOSCAN_LOG_DEBUG("  - Client streaming: {}", client_->isStreaming() ? "YES" : "NO");
+          HOLOSCAN_LOG_DEBUG("  - Upstream ready: {}", client_->isUpstreamReady() ? "YES" : "NO");
+
+          send_success = client_->sendFrame(frame);
+          if (send_success) {
+            HOLOSCAN_LOG_INFO("✅ Frame sent successfully on attempt {}",
```

Comment:
I am not so sure of using icons and emojis in the code. Having a series of tests that provides a summary of what has passed or failed would be much cleaner and easier to maintain? @tbirdso what do you think?